app/connectors/IndividualDetailsConnector.scala

```diff
@@ -47,19 +47,24 @@ class DefaultIndividualDetailsConnector @Inject() (httpClient: HttpClient,
   def getIndividualDetails(identifier: IndividualDetailsIdentifier, resolveMerge: ResolveMerge
                           )(implicit ec: ExecutionContext,hc: HeaderCarrier, correlationId: CorrelationId
   ): IndividualDetailsResponseEnvelope[IndividualDetails] = {
-    val url = s"${appConfig.individualDetailsServiceUrl}/individuals/details/NINO/${identifier.value}/${resolveMerge.value}"
-    val connectorName     = "individual-details-connector"
-    val additionalLogInfo = Some(AdditionalLogInfo(Map("correlation-id" -> correlationId.value.toString)))
-    withHttpReads(
-      connectorName,
-      metrics.defaultRegistry,
-      additionalLogInfo
-    ) { implicit httpReads =>
-      EitherT(
-        httpClient
-          .GET(url)(httpReads, desApiHeaders(appConfig.individualDetails), ec)
-          .recovered(logger, connectorName, metrics.defaultRegistry, additionalLogInfo)
-      )
+
+    if (identifier.value.isEmpty) {
+      IndividualDetailsResponseEnvelope(Left(ConnectorError(400, "Cannot call getIndividualDetails with an empty identifier")))
```

Comment:
I am assuming that this is handled by the caller and the block correctly routes based on the fact this was not matched?

Reply:
Also do we need to add a test for this?

Reply:
@rishisachania  @rambabu-posa  - the more I think about this the more I do not like this approach. This is not a connector error. It's not an error at all. It's simply that we do not want to attempt to make this call if we have no Nino. Triggering an error is misleading and I think we need a different approach. Let me think about this.